common/persistence/dataInterfaces.go

```diff
@@ -511,20 +521,22 @@ type (
 		FinishedExecutionTTL int32
 
 		// Mutable state
-		UpsertActivityInfos       []*ActivityInfo
-		DeleteActivityInfos       []int64
-		UpserTimerInfos           []*TimerInfo
-		DeleteTimerInfos          []string
-		UpsertChildExecutionInfos []*ChildExecutionInfo
-		DeleteChildExecutionInfo  *int64
-		UpsertRequestCancelInfos  []*RequestCancelInfo
-		DeleteRequestCancelInfo   *int64
-		UpsertSignalInfos         []*SignalInfo
-		DeleteSignalInfo          *int64
-		UpsertSignalRequestedIDs  []string
-		DeleteSignalRequestedID   string
-		NewBufferedEvents         *SerializedHistoryEventBatch
-		ClearBufferedEvents       bool
+		UpsertActivityInfos           []*ActivityInfo
+		DeleteActivityInfos           []int64
+		UpserTimerInfos               []*TimerInfo
+		DeleteTimerInfos              []string
+		UpsertChildExecutionInfos     []*ChildExecutionInfo
+		DeleteChildExecutionInfo      *int64
+		UpsertRequestCancelInfos      []*RequestCancelInfo
+		DeleteRequestCancelInfo       *int64
+		UpsertSignalInfos             []*SignalInfo
+		DeleteSignalInfo              *int64
+		UpsertSignalRequestedIDs      []string
+		DeleteSignalRequestedID       string
+		NewBufferedEvents             *SerializedHistoryEventBatch
+		ClearBufferedEvents           bool
+		NewBufferedReplicationTask    *BufferedReplicationTask
```

Comment:
should we use a slice? or is NewBufferedReplicationTask and DeleteBufferedReplicationTask guaranteed to have at most one at a time?

Reply:
Currently we are applying replication task as a single transaction, so there is no need currently.  We will cross that bridge when there is a need for batching them together.